baseband/dada/base.py

```diff
@@ -268,56 +265,21 @@ class DADAStreamWriter(DADAStreamBase, VLBIStreamWriterBase, DADAFileWriter):
     def __init__(self, fh_raw, header, squeeze=True):
         assert header.get('OBS_OVERLAP', 0) == 0
         super(DADAStreamWriter, self).__init__(fh_raw, header, squeeze=squeeze)
-        self._frame = self.memmap_frame(header)
-        self._frame_nr = 0
-
-    def write(self, data, invalid_data=False):
-        """Write data, using multiple files as needed.
 
-        Parameters
-        ----------
-        data : array
-            Piece of data to be written, with sample dimensions as given by
-            ``sample_shape``. This should be properly scaled to make best use
-            of the dynamic range delivered by the encoding.
-        invalid_data : bool, optional
-            Whether the current data is valid.  Defaults to `False`.  Present
-            for consistency with other stream writers.  It does not seem
-            possible to store this information in DADA files.
-        """
-        assert data.shape[1:] == self.sample_shape, (
-            "'data' should have trailing shape {}".format(self.sample_shape))
-        if self.squeeze:
-            data = self._unsqueeze(data)
-
-        count = data.shape[0]
-        sample = 0
-        offset0 = self.offset
-        while count > 0:
-            frame_nr, sample_offset = self._frame_info()
-            if self._frame_nr is None:
-                assert sample_offset == 0
-                self._get_frame(frame_nr)
-
-            nsample = min(count, self.samples_per_frame - sample_offset)
-            sample_end = sample_offset + nsample
-            sample = self.offset - offset0
-            self._frame[sample_offset:
-                        sample_end] = data[sample:sample + nsample]
-            if sample_end == self.samples_per_frame:
-                # deleting frame flushes memmap'd data to disk
-                del self._frame
-                self._frame_nr = None
-
-            self.offset += nsample
-            count -= nsample
-
-    def _get_frame(self, frame_nr):
+    def _make_frame(self, index):
```

Comment:
OK, `_make_frame` it is!

Reply:
Just to throw it out there: `_new_frame`?

Reply:
Not that `_new_frame` isn't tempting, but I like the pairing of `_make_frame` and `_write_frame` (I suppose because the first word is a verb for both).